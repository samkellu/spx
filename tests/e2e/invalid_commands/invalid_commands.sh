#! /bin/bash
RED='\033[0;31m'
GREEN='\033[0;34m'
YELLOW='\033[0;33m'
NC='\033[0m'

test_name="invalid_commands"

gcc tests/E2E/$test_name/test_trader.c -o tests/E2E/$test_name/test_trader -Wall -Werror -Wvla -O0 -std=c11 -g -fsanitize=address,leak -D_POSIX_C_SOURCE=199309L -lm

./spx_exchange "products.txt" "tests/E2E/$test_name/test_trader" | tee tests/E2E/$test_name/$test_name.test

diff tests/E2E/$test_name/$test_name.out tests/E2E/$test_name/$test_name.test && echo -e "${GREEN} $test_name Exchange output correct.${NC}" || echo -e "${RED} $test_name Exchange output incorrect.${NC}"

diff tests/E2E/$test_name/trader.out tests/E2E/$test_name/trader.test && echo -e "${GREEN} $test_name Trader output correct.${NC}" || echo -e "${RED} $test_name Trader output incorrect.${NC}"

gcov ./spx_exchange

rm tests/E2E/$test_name/$test_name.test
rm tests/E2E/$test_name/trader.test
