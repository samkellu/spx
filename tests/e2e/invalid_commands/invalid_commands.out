[SPX] Starting
[SPX] Trading 2 products: GPU Router
[SPX] Created FIFO /tmp/spx_exchange_0
[SPX] Created FIFO /tmp/spx_trader_0
[SPX] Starting trader 0 (tests/E2E/invalid_commands/test_trader)
[SPX] Connected to /tmp/spx_exchange_0
[SPX] Connected to /tmp/spx_trader_0
[SPX] [T0] Parsing command: <INVALID>
[SPX] [T0] Parsing command: <SELL -1 GPU 10 10>
[SPX] [T0] Parsing command: <SELL 1 GPU 30 20>
[SPX] [T0] Parsing command: <SELL 0 DOG 100 100>
[SPX] [T0] Parsing command: <SELL 0 GPU -1 100>
[SPX] [T0] Parsing command: <SELL 0 GPU 10 -100>
[SPX] [T0] Parsing command: <SELL 0 GPU 1000000 100>
[SPX] [T0] Parsing command: <SELL 0 GPU 100 1000000>
[SPX] [T0] Parsing command: <SELL 0 GPU 1000000 100CANCEL 12>
[SPX] [T0] Parsing command: <AMEND 22 22 22>
[SPX] Trader 0 disconnected
[SPX] Trading completed
[SPX] Exchange fees collected: $0
